core/server/models/plugins/filter.js

```diff
@@ -85,6 +85,18 @@ filter = function filter(Bookshelf) {
         },
 
         preProcessFilters: function preProcessFilters() {
+            /* Lowercase filters for tags and slug no matter how they are passed in */
+            this._filters.statements = gql.json.replaceStatements(this._filters.statements, {prop: /tags|slug/}, function (statement) {
```

Comment:
Looks like the right place to add 👍 

What about authors/users queries containing uppercase slugs?

e.g.
> {{#get "users" filter="slug:TEST"}}

Reply:
It has to 👍 I haven't checked out your PR. Could you pls add one test case which proofs that it works for users/authors? Thanks :)

Reply:
The unit test is at the filter level which if this filter is applied to all models. My tests currently don't test Post model either. I currently don't see any integration tests to work at that level but let me look around. If anyone could point me in the correct direction that would be awesome.

Reply:
I was talking about adding e.g.

> {prop: 'authors', op: '=', value: 'TEST-123'}
> {prop: 'author', op: '=', value: 'TEST-123'}
> {prop: 'user', op: '=', value: 'TEST-123'}

Reply:
Sorry about the delay are you sure we want to remove case from all of these? It expands the issue that this PR was fixing.

Reply:
It not only affects the tags. It's weird only fixing it for tags, don't you think?

Let me know if you have time, otherwise i'll add some extra commits on top of this PR :)

Reply:
This just does it for tags or slug which from my understanding was always suppose to be lowercase and other fields are not forced lowercase.

Reply:
Will look at the PR next Monday again 👍

Reply:
Sorry for late reply.

I shared the wrong example

> {{#get "posts" filter="author:TOMAS"}}
> {{#get "posts" filter="primary_author:TOMAS"}}

This use case won't get covered, because your filter by props "slug" or "tags".

Reply:
Users/Authors have a slug too.

e.g. `filter="author:TOMAS"` refers to the user slug.

If i serve `/author/THOMAS/`, it's the slug too. Authors are served via the slug.

**Each slug in the Ghost system is lower case and runs through a [safe function.](https://github.com/TryGhost/Ghost/blob/2.6.2/core/server/lib/security/string.js)**

Reply:
Then wouldn't the filter field be called slug and not author?

Reply:
There are a couple of short cuts e.g. primary_author, author, see [docs](https://api.ghost.org/docs/filter#simple-examples)

I'd suggest the following: we wait till #10159 was merged (it is heavily changing the filter plugin) and i will afterwards update your PR. How does that sound to you? 🙂

Reply:
Ok I am always game for you to make any changes need to get this going but I let me rebase and see if there is some way for me to figure out if the property is a shortcut to a tag or slug and to replace those as well.

Reply:
Sorry for the rebase ☹️We replaced GQL and had to refactor the filter plugin.
Let me know if you need a hand, i can help 👍